CRM/Civicase/Event/Listener/CaseTypeCategoryIsActiveToggler.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+use CRM_Civicase_Factory_CaseTypeCategoryEventHandler as CaseTypeCategoryEventHandlerFactory;
+use Civi\API\Event\RespondEvent;
+
+/**
+ * Class CRM_Civicase_Event_Listener_CaseTypeCategoryIsActiveToggler.
+ *
+ * Contains extra processing for case type category when it is
+ * enabled / disabled, e.g. show / hide related menu item.
+ */
+class CRM_Civicase_Event_Listener_CaseTypeCategoryIsActiveToggler {
+
+  /**
+   * Update case type category related objects on category update.
+   *
+   * @param \Civi\API\Event\RespondEvent $e
+   *   Event data.
+   */
+  public static function onRespond(RespondEvent $e) {
+    $apiRequest = $e->getApiRequest();
+    if ($apiRequest['entity'] == 'OptionValue' && $apiRequest['action'] == 'create') {
```

Comment:
HTTP Referrer checks is needed in this class only and should be made in this class. The logic for that should be moved from the event handler class to this class. If we don't do that, the Event handler class will get bloated with un-necessary checks

Reply:
In case if we need this somewhere else we have to duplicate code from one listener/hook processing to another. Why not to have it one place? EventHandler class souns suitable for having such methods, so why not?

Reply:
In that case the check need to be moved to a central function so that the listener class and any other class will use it from there. It is not the responsibility of the handler class to perform these checks.